rustc-wrapper.c

```diff
@@ -0,0 +1,64 @@
+#include <unistd.h>
+#include <fcntl.h>
+#include <string.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/wait.h>
+
+int main(int argc, char *argv[]) {
+  char **args = calloc(argc + 1, sizeof(char *));
+  const char *file = NULL;
+  int nargs = 1;
+  for (int i = 1; i < argc; i++) {
+    if (strcmp(argv[i], "--notify") == 0) {
+        file = argv[i + 1];
+        i++;
+    } else {
+        args[nargs++] = argv[i];
+    }
+  }
+  args[0] = "rustc";
```

Comment:
This won't respect machine file overrides, I guess... ;)

Probably should take it as a CLI argument.